src/main/java/graphql/schema/GraphQLEnumType.java

```diff
@@ -96,6 +96,12 @@ private Object getNameByValue(Object value) {
             if (value.equals(definitionValue)) {
                 return valueDefinition.getName();
             }
+            // we can treat enum backing values as strings in effect
+            if (definitionValue instanceof Enum  && value instanceof String) {
+                if (value.equals(definitionValue.toString())) {
```

Comment:
Wouldn't it be better to call `name()` instead? `toString()` may be overriden by owner of the enum class whereas `name()` is final.

Reply:
yes it would